src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/GerritProject.java

```diff
@@ -210,23 +212,24 @@ public void setForbiddenFilePaths(List<FilePath> forbiddenFilePaths) {
      * @param branch the branch.
      * @param topic the topic.
      * @param files the files.
+     * @param envVars the environment variables exisiting on the jenkins host.
      * @return true is the rules match.
      */
-    public boolean isInteresting(String project, String branch, String topic, List<String> files) {
-        if (compareType.matches(pattern, project)) {
+    public boolean isInteresting(String project, String branch, String topic, List<String> files, EnvVars envVars) {
```

Comment:
Same here. This is a public api and needs to keep backwards compatibility as much as possible.

Keep the old signature, deprecate it and have it call the new signature with `null` to the new parameter.